### PR TITLE
Simple pluggable serve-and-handle API

### PR DESCRIPTION
**Pull request by tucnak**
This is my take on serve-and-handle API, which I think we can introduce effortlessly. It doesn't brake any existing API. It also uses radix tree matching for path serving. Here's the sample of proposed API:

```go
bot, err := telebot.NewBot(os.Getenv("BOT_TOKEN"))
if err != nil {
	log.Fatalln(err)
}

bot.Handle("/start", func (c telebot.Context) {
	b, m := c.Bot, c.Message

	b.SendMessage(m.Sender, "Hello!", nil)
})

messages := make(chan telebot.Message, 100)
bot.Listen(messages, 1*time.Second)

for message := range messages {
	bot.Serve(&message)
}
```

### CODE REVIEW

**Comment by irgendwr** [comment]
Shouldn't `bot.Serve(&message)` be `bot.Serve(message)` after f7a78ef?

File: handlers.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package telebot
+
+// Default handler prefix.
+const Default string = ""
```
Comment (irgendwr) [inline handlers.go]: This constant is never used

Comment (tucnak) [reply handlers.go]: I added it for convenience, so I could write:
```go
b.Handle(tb.Default, func(c Context) {})
```

Comment (irgendwr) [reply handlers.go]: Ah makes sense.

**Pull request merged**